knn/knn.h

```diff
@@ -59,6 +59,8 @@ struct ModelSettings_t
 	std::string m_sModelName;
 	std::string m_sCachePath;
 	std::string m_sAPIKey;
+	std::string m_sAPIUrl;
+	int			m_iAPITimeout = 0; // 0 means use default (10 seconds), positive value is timeout in seconds
```

Comment:
This is confusing. Why do we set 0 and say 0 means default 10 seconds? If we set the default to 10, we should set 10; 0 is the common pattern for no limit at all.